module/zfs/zfs_fm.c

```diff
@@ -1444,6 +1444,60 @@ zfs_ereport_fini(void)
 	mutex_destroy(&recent_events_lock);
 }
 
+void
+zfs_ereport_snapshot_post(const char *subclass, spa_t *spa, const char *name)
+{
+	nvlist_t *aux;
+
+	aux = fm_nvlist_create(NULL);
+	nvlist_add_string(aux, "snapshot_name", name);
+
+	zfs_post_common(spa, NULL, FM_RSRC_CLASS, subclass, aux);
+	fm_nvlist_destroy(aux, FM_NVA_FREE);
+}
+
+/*
+ * Post when a event when a zvol is created or removed
+ *
+ * This is currently only used by macOS, since it uses the event to create
+ * symlinks between the volume name (mypool/myvol) and the actual /dev
+ * device (/dev/disk3).  For example:
+ *
+ * /var/run/zfs/dsk/mypool/myvol -> /dev/disk3
+ *
+ * name: The full name of the zvol ("mypool/myvol")
+ * dev_name: The full /dev name for the zvol ("/dev/disk3")
+ * raw_name: The raw  /dev name for the zvol ("/dev/rdisk3")
+ */
+void
+zfs_ereport_zvol_post(const char *subclass, const char *name,
+    const char *dev_name, const char *raw_name)
+{
+	nvlist_t *aux;
+	char *r;
+	spa_t *spa;
+	boolean_t has_lock = B_FALSE;
+
+	has_lock = mutex_owned(&spa_namespace_lock);
+	if (!has_lock) mutex_enter(&spa_namespace_lock);
+	spa = spa_lookup(name);
```

Comment:
```suggestion
	spa_t *spa = spa_lookup(name);
```

nit: we've gradually been moving to declaring the variables at first use when it makes sense.